pkg/clusteragent/clusterchecks/stores.go

```diff
@@ -102,3 +116,71 @@ func (s *nodeStore) removeConfig(digest string) {
 	delete(s.digestToConfig, digest)
 	dispatchedConfigs.WithLabelValues(s.name).Dec()
 }
+
+// AddRunnerStats stores runner stats for a check
+// The nodeStore handles thread safety for this public method
+func (s *nodeStore) AddRunnerStats(checkID string, stats types.CLCRunnerStats) {
+	s.Lock()
+	defer s.Unlock()
+	s.clcRunnerStats[checkID] = stats
+}
+
+// RemoveRunnerStats deletes runner stats for a check
+// The nodeStore handles thread safety for this public method
+func (s *nodeStore) RemoveRunnerStats(checkID string) {
+	s.Lock()
+	defer s.Unlock()
+	_, found := s.clcRunnerStats[checkID]
+	if !found {
+		log.Debugf("unknown check ID %s, skipping", checkID)
+		return
+	}
+	delete(s.clcRunnerStats, checkID)
+}
+
+// GetRunnerStats returns the runner stats of a given check
+// The nodeStore handles thread safety for this public method
+func (s *nodeStore) GetRunnerStats(checkID string) types.CLCRunnerStats {
```

Comment:
do you think it can be useful that `GetRunnerStats` return also an error in case of `checkID` not found?

Reply:
totally agree